# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBCClass.h
# Commit: 349728105c40
# Full Hash: 349728105c40993896170f4c2f43e152a3466706
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2022-01-08 09:40:24
# Description:
#   Bug 1748602 - Backout bug 1642412. r=lth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D135346
# ==============================================================================

diff -r 32873022d2ed -r 349728105c40 js/src/wasm/WasmBCClass.h
--- a/js/src/wasm/WasmBCClass.h	Fri Jan 07 20:35:15 2022 +0000
+++ b/js/src/wasm/WasmBCClass.h	Fri Jan 07 20:43:16 2022 +0000
@@ -1070,15 +1070,6 @@
 
   //////////////////////////////////////////////////////////////////////
   //
-  // Table access.
-
-  Address addressOfTableField(const TableDesc& table, uint32_t fieldOffset,
-                              RegPtr tls);
-  void loadTableLength(const TableDesc& table, RegPtr tls, RegI32 length);
-  void loadTableElements(const TableDesc& table, RegPtr tls, RegPtr elements);
-
-  //////////////////////////////////////////////////////////////////////
-  //
   // Heap access.
 
   void bceCheckLocal(MemoryAccessDesc* access, AccessCheck* check,
@@ -1534,10 +1525,6 @@
   [[nodiscard]] bool emitTableSet();
   [[nodiscard]] bool emitTableSize();
 
-  void emitTableBoundsCheck(const TableDesc& table, RegI32 index, RegPtr tls);
-  [[nodiscard]] bool emitTableGetAnyRef(uint32_t tableIndex);
-  [[nodiscard]] bool emitTableSetAnyRef(uint32_t tableIndex);
-
 #ifdef ENABLE_WASM_GC
   [[nodiscard]] bool emitStructNewWithRtt();
   [[nodiscard]] bool emitStructNewDefaultWithRtt();